### DISCUSSION TITLE
Enhanced Diagnostic Information

### DISCUSSION THREAD

**Discussion by baxgit** [revk/xESP32-Faikout#448] [Ideas]
Firstly, thank you for producing "Faikin" - I have two of the units and they work brilliantly. 

I'd like to suggest making available enhanced diagnostic information sourced from the AC unit. For example, I'm using Faikin with a FDXM50F9 unit linked to Home Assistant. Home Assistant automatically creates diagnostic sensors for: Fan Frequency; Inlet Air Temperature; and Coolant Liquid Temperature. However, the Daikin (bluetooth) app that I have and which connects to my indoor unit via a BRC1H52W wall controller, can "see" there is other diagnostic data potentially available, e.g. Gas Pipe Temperature, Ant-freeze Control, Fan Speed etc. 

I appreciate diagnostic data may vary by Daikin unit, but it would be extremely helpful to have access to additional diagnostic sensor information if available. Thanks.

**Comment by revk** [comment]
X50A or S21?

**Reply by baxgit** [reply]
X50A, though I’m hoping to add a couple of S21 based units

**Reply by revk** [reply]
Well the good news is I have an X50A interface. I'll look in to it. I think we do get those details.

**Reply by revk** [reply]
Unlikely to be today, another big project on...

**Comment by revk** [comment]
OK fan speed is reported already.

I don't see a gas pipe temperature, I have a `liquid` status which might be the same thing. If not, then do you know where it is in the X50A message?

`antifreeze` status is an on/off, as is `slave` (i.e. cannot do heat/cool as other device is opposite), also some `flap` status.

Do you know an HA thing for a simple on/off status? If you can let me know I can add those.

**Reply by baxgit** [reply]
Great. I expect things vary by unit, but for mine there's also a fan 'RPM' if that's available in the X50A message?

I've attached screen shots from the Daikin app (for what it's worth) which refers to a 'Liquid Pipe' and a 'Gas Pipe' temperature. Afraid I can't help re the X50A message - just out of interest, what do you use to read and analyse the X50A signal output from the Daikin unit?

Re HA, I'd normally use a binary_sensor for a simple on/off status.

![IMG_8078 (Small)](https://github.com/user-attachments/assets/a092d984-57c6-4166-b907-3b02659edfcf)
![IMG_8080 (Small)](https://github.com/user-attachments/assets/ba5507ad-313e-4d56-a3ae-45f171a9cd31)

**Comment by revk** [comment]
I see fan RPM, but as you say it may vary by model.

<img width="370" alt="Screenshot 2024-08-25 at 07 11 50" src="https://github.com/user-attachments/assets/6b02d004-72e0-4de0-91a8-2de29b79848b">

**Comment by serimi89** [comment]
I believe these are messages that are passed down via the P1 protocol only. The Madoka app that you are using connects via Bluetooth to the BRC1H52W controller which is then connected through the P1 port to the PCB of the AC. 
The Faikin (or the Daikin WiFi adaptor) then sits on the X50A port, however they dont share the same information with the P1 port.

For example: The BRC1H52W controller has an 'external' temp sensor which can be programmed to be the one controlling the AC. The value of this temperature sensor is not visible when using the Faikin (or the Daikin WiFi adaptor) - only the internal one will be visible.
Also, many Field Settings can be manipulated via the P1 port - I am currently not aware that this is possible via the X50A / S21 protocol.

**Reply by revk** [reply]
Yeh, the P1 connection definitely has way more than the WiFi module did, including a load of extra settings.

**Reply by baxgit** [reply]
OK, that makes sense

**Reply by serimi89** [reply]
On that note - I found it its more convenient to control these ducted units (like yours) via the Madoka controller and a ESP to BLE connection. There is an integration for ESPHome to Madoka connection. 
As mentioned above in that way you read the actual room temp from the Madoka. I would like to see some more diagnostics reported, but maybe that's something for the future.
Sorry @revk 😄 Faikin is not ALWAYS the solution haha
